src/Components/Aspire.Azure.Messaging.WebPubSub/AspireWebPubSubExtensions.cs

```diff
@@ -0,0 +1,138 @@
+
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Microsoft.Extensions.Hosting;
+
+/// <summary>
+/// Provides extension methods for registering keyed service <see cref="WebPubSubServiceClient"/> as a singleton with hubName as the key in the services provided by the <see cref="IHostApplicationBuilder"/>.
+/// </summary>
+public static class AspireWebPubSubExtensions
+{
+    private const string DefaultConfigSectionName = "Aspire:Azure:Messaging:WebPubSub";
+
+    /// <summary>
+    /// Registers a keyed service <see cref="WebPubSubServiceClient"/> as a singleton with hubName as the key in the services provided by the <paramref name="builder"/>.
+    /// Access the <see cref="WebPubSubServiceClient"/> using <see cref="FromKeyedServicesAttribute"/> with hubName as the key.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="hubName">The hub used for Web PubSub service, hubName is also considered as the name of the registered <see cref="WebPubSubServiceClient"/> as well as the serviceKey of the service</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="AzureMessagingWebPubSubSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientBuilder">An optional method that can be used for customizing the <see cref="IAzureClientBuilder{WebPubSubServiceClient, WebPubSubServiceClientOptions}"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire.Azure.Messaging.WebPubSub" section.</remarks>
+    /// <exception cref="InvalidOperationException">Thrown when neither <see cref="AzureMessagingWebPubSubSettings.ConnectionString"/> nor <see cref="AzureMessagingWebPubSubSettings.Endpoint"/> is provided.</exception>
+    /// <example>
+    /// Use the following code to add a Web PubSub service client to the services collection:
+    /// <code>
+    /// builder.AddAzureWebPubSubHub("wps1", "chatHub");
+    /// </code>
+    /// </example>
+    /// <example>
+    /// Use the following code to use the registered service client in a controller:
+    /// <code>
+    /// public class HomeController : Controller
+    /// {
+    ///     private readonly WebPubSubServiceClient _service;
+    ///     public HomeController([FromKeyedServices("chatHub")]WebPubSubServiceClient service)
+    ///     {
+    ///         _service = service;
+    ///     }
+    /// }
+    /// </code>
+    /// </example>
+    public static void AddAzureWebPubSubHub(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        string hubName,
```

Comment:
I think we should follow the same API / design pattern that is used in EventHubs:

1. The EventHubName is flown through the Settings class:
https://github.com/dotnet/aspire/blob/ff9732fafed47140ac38226acdfef4ecd8eb5942/src/Components/Aspire.Azure.Messaging.EventHubs/AzureMessagingEventHubsSettings.cs#L39-L42
This way, it automatically can be configured to be a different name, rather than being hard-coded in the callsite.
2. Ensure the hub name is specified:
https://github.com/dotnet/aspire/blob/ff9732fafed47140ac38226acdfef4ecd8eb5942/src/Components/Aspire.Azure.Messaging.EventHubs/EventHubsComponent.cs#L98-L104
3. Then have the common pattern of `AddAzureWebPubSubServiceClient` and `AddKeyedAzureWebPubSubServiceClient`.

(note I think we should change the name of the method to align with the class name: `WebPubSubServiceClient`)

Reply:
The key difference between webpubsub and event hub is, for webpubsub, different hubs share the same connection string,  however different hubs use different `WebPubSubServiceClient`. The key difference `hub` in webpubsub is super light-weighted, as long as a client connects to a hub then the hub exists, it is kind of a "session" logic. there is no need to "create" or "deploy" a hub in the control plane, it means that there is no "provisioning" step for a "hub" in webpubsub, so we should only have one "AddWebPubSub" in provisioning stage to create one WebPubSub service. In control plane there is one "HubSettings" subresource, but this is just some settings for a hub and is not required.

Reply:
sounds cool

Reply:
I think the only gap is how to add different serviceClient while sharing the same ConnectionString.

In WebPubSub hubs are lightweighted concept that there is no need to explicitly "create" or "provision" a hub from the control plane, it is kind of some session concept that a client always connects to a hub. So different hubs share the same connection string, but use different `WebPubSubServiceClient` to provide different paths, e.g. `/client/hubs/{hub}` to the client. 

Consider an app server containing 2 different hubs `chat` and `notification`:
* Step1, AppHost provisions the WebPubSub resource using `builder.AddAzureWebPubSub("wps1")`
* Step2, the app server exposes different endpoints for different hubs, for example, `app.MapGet("/chat", (ServiceClientForHubChat client)=>{})`, `app.MapGet("/notification", (ServiceClientForNotificationHub client)=>{})` 

If using API similar to `AddKeyedWebPubSubServiceClient("wps1", o=>o.HubName="chat")`, how different hubs could consume the same WebPubSub service connection string while register different keyed serviceClient? And how different hubs could have different `ClientOptions` reading from settings? Currently `AddKeyed...ServiceClient` uses the name as the connectionName, the registered serviceClient name, and also the configuration section name.

I am separating `name` into `connectionName` and `serviceKey`, and configuration to read from `{connectionName}:{serviceKey}`, how to you feel about this approach @eerhardt ?

Reply:
I like this approach. Thanks for updating it.